src/snowflake/cli/_plugins/nativeapp/artifacts.py

```diff
@@ -658,6 +667,11 @@ def build_bundle(
     for artifact in artifacts:
         bundle_map.add(artifact)
 
+    if bundle_map.is_empty():
```

Comment:
for drop, wasn't part of the requirement to allow it to go through?

Reply:
Correct. I don't think we should raise this exception in build_bundle, we should return the empty map instead. Individual uses of the bundle map can then check their preconditions as needed.

Reply:
Though I had mistakenly included `drop` as a command that invokes build_bundle (it's not even a command :')), `teardown` (which is what we mean here) does not invoke `build_bundle`. 

I think it might be fair to say that any command that calls `build_bundle` effectively needs the bundle to not be empty, (and `teardown` does not call build_bundle). Having said that, it should make sense to keep the is_empty check as part of build_bundle. 

If we foresee that we may have a use case for calling build_bundle and allowing an empty bundle, I can add a more specific exception class for  "empty bundle", and then we can catch that specific exception in the caller and ignore it. 

What do you think?

Reply:
Below two screenshots show:
`run` command with artifacts succeeds. Then artifacts are commented out (empty), `teardown` command succeeds:
![run_with](https://github.com/user-attachments/assets/89827f1f-bfc7-47a4-8ae4-401af703f6c6)
![teardown_wo](https://github.com/user-attachments/assets/018ce091-a313-4f51-9b58-0030d585ee66)